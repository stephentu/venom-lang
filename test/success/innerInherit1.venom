class Baz
  attr x::string
  def self(x::string) = 
    print(x); 
    self.x = x;
  end
  def getX() -> string = x; end
end

class A
  class B <- Baz
    def self(x::int, y::string) : super(y) = print(x); end
  end

  attr elems::list{B} = []

  def addB(x::int, y::string) =
    b = B(x, y);
    elems.append(b);
  end
end

a = A();
a.addB(10, "HI");
a.addB(20, "BYE");

print(a.elems[0].getX());
print(a.elems[1].getX());
