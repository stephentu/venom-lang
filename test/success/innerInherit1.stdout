HI
10
BYE
20
HI
BYE
